app/forms/reactivate_profile_form.rb

```diff
@@ -9,9 +9,11 @@ class ReactivateProfileForm
   attr_accessor :recovery_code, :password
   attr_reader :user
 
-  def initialize(user, attrs = {})
+  def initialize(user, attrs = { recovery_code: [] })
```

Comment:
Ah, I understand the goal now. I still think it should be `attrs = {}` because you could pass in a hash with `password` but not `recovery_code` and then recovery code would still be nil, not an array.

I think this would be clearer with like a `attrs[:recovery_code] ||= []` in the first line before we call super, that will make sure we don't have nil